buzsaki_lab_to_nwb/grosmark_code/grosmarklfpdatainterface.py

```diff
@@ -0,0 +1,49 @@
+"""Authors: Cody Baker and Ben Dichter."""
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from pynwb import NWBFile
+import os
+import warnings
+
+from ..neuroscope import read_lfp, write_lfp, write_spike_waveforms
+
+
+class GrosmarkLFPInterface(BaseDataInterface):
+    """Primary data interface for LFP aspects of the GrosmarkAD dataset."""
+
+    @classmethod
+    def get_input_schema(cls):
+        """Return subset of json schema for informing the NWBConverter of expepcted input arguments."""
+        return dict(properties=dict(folder_path="string"))
+
+    def convert_data(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        """Convert the LFP portion of a particular session of the GrosmarkAD dataset."""
+        session_path = self.input_args['folder_path']
+        all_shank_channels = metadata['all_shank_channels']
+        lfp_sampling_rate = metadata['lfp_sampling_rate']
+        spikes_nsamples = metadata['spikes_nsamples']
+        shank_channels = metadata['shank_channels']
+        n_total_channels = metadata['n_total_channels']
```

Comment:
Why can't you pull these values here?

Reply:
The current format is somewhat legacy to reduce the number of times the .xml has to be read from (not a terribly expensive computation, I'll admit). Basically, all of these fields are used in the metadata of the electrode table so under this design the NWBConverter accesses the .xml once, pulls that info, and passes it to both the RecordingExtractorInterface and the LFPInterface, but with differently keyed metadata since it's still per-interface here.

Once the new changes go through of course, all metadata will occur only once in the total metadata object and the interfaces will simple find it there, or pull what they need within their own interface.

Reply:
@CodyCBakerPhD Can you change it so that this DataInterface pulls this metadata directly from the xml? Is there some pending change that is blocking that?

Reply:
@bendichter Sure, I can change that now.

Reply:
Resolved on this interface